sdk/storage/src/common/storage_error.cpp

```diff
@@ -4,92 +4,120 @@
 #include "common/storage_error.hpp"
 
 #include "common/xml_wrapper.hpp"
-#include "http/http.hpp"
+
+#include <type_traits>
 
 namespace Azure { namespace Storage {
-  StorageError StorageError::CreateFromResponse(/* const */ Azure::Core::Http::Response& response)
+  StorageError StorageError::CreateFromResponse(
```

Comment:
I think we'll want to mimic what .NET has done here.
A RequestFailedException class in Core similar to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/RequestFailedException.cs

and then a StorageRequestFailedException class derived from it that is similar to this: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/StorageRequestFailedException.cs

Reply:
The members within the .NET SDK are similar to the current design. `StorageError` is currently derived from `std::runtime_error` due to the `RequestFailedException` not ready yet in Azure core and we don't think it is a priority.
I have created an issue to track this change in the future when `RequestFailedException` is ready.
https://github.com/Azure/azure-sdk-for-cpp/issues/205